src/sentry/mail/adapter.py

```diff
@@ -140,62 +149,82 @@ def should_notify(self, group):
-                if teams_to_resolve:
-                    send_to_list |= set(
-                        User.objects.filter(
-                            is_active=True,
-                            sentry_orgmember_set__organizationmemberteam__team__id__in=teams_to_resolve,
-                        ).values_list("id", flat=True)
-                    )
-
-                alert_settings = project.get_member_alert_settings(self.alert_option_key)
-                disabled_users = set(
-                    user for user, setting in alert_settings.items() if setting == 0
-                )
-                return send_to_list - disabled_users
+        send_to = []
+        if target_type == ActionTargetType.ISSUE_OWNERS:
+            if not event:
+                send_to = self.get_send_to_all_in_project(project)
             else:
+                send_to = self.get_send_to_owners(event, project)
+        elif target_type == ActionTargetType.MEMBER:
+            send_to = self.get_send_to_member(target_identifier)
+        elif target_type == ActionTargetType.TEAM:
+            send_to = self.get_send_to_team(project, target_identifier)
+        return set(send_to)
+
+    def get_send_to_owners(self, event, project):
+        owners, _ = ProjectOwnership.get_owners(project.id, event.data)
+        if owners != ProjectOwnership.Everyone:
+            if not owners:
                 metrics.incr(
                     "features.owners.send_to",
-                    tags={"organization": project.organization_id, "outcome": "everyone"},
+                    tags={"organization": project.organization_id, "outcome": "empty"},
                     skip_internal=True,
                 )
+                return set()
 
+            metrics.incr(
+                "features.owners.send_to",
+                tags={"organization": project.organization_id, "outcome": "match"},
+                skip_internal=True,
+            )
+            send_to = set()
+            teams_to_resolve = set()
+            for owner in owners:
+                if owner.type == User:
+                    send_to.add(owner.id)
+                else:
+                    teams_to_resolve.add(owner.id)
+
+            # get all users in teams
+            if teams_to_resolve:
+                send_to |= set(
+                    User.objects.filter(
+                        is_active=True,
+                        sentry_orgmember_set__organizationmemberteam__team__id__in=teams_to_resolve,
+                    ).values_list("id", flat=True)
+                )
+
+            return send_to - self.disabled_users_from_project(project)
+        else:
+            return self.get_send_to_all_in_project(project)
+
+    def disabled_users_from_project(self, project):
+        alert_settings = project.get_member_alert_settings(self.alert_option_key)
+        disabled_users = set(user for user, setting in alert_settings.items() if setting == 0)
+        return disabled_users
```

Comment:
Note that this logic is mostly just straight extracted from `get_send_to`, but there are some minor changes